vars/seleniumExecuteTests.groovy

```diff
@@ -102,11 +101,9 @@ void call(Map parameters = [:], Closure body) {
 
         // Inject config via env vars so that scripts running inside selenium can respond to that
         config.dockerEnvVars = config.dockerEnvVars ?: [:]
-        config.dockerEnvVars.PIPER_CONTAINER_PORT_MAPPING = new JsonOutput().toJson(config.containerPortMappings)
-        config.dockerEnvVars.PIPER_DOCKER_NAME = new JsonOutput().toJson(config.dockerName)
-        config.dockerEnvVars.PIPER_DOCKER_IMAGE = new JsonOutput().toJson(config.dockerImage)
-        config.dockerEnvVars.PIPER_SIDECAR_NAME = new JsonOutput().toJson(config.sidecarName)
-        config.dockerEnvVars.PIPER_SIDECAR_IMAGE = new JsonOutput().toJson(config.sidecarImage)
+        config.dockerEnvVars.PIPER_SELENIUM_HOSTNAME = config.dockerName
+        config.dockerEnvVars.PIPER_SELENIUM_WEBDRIVER_HOSTNAME = config.sidecarName
+        config.dockerEnvVars.PIPER_SELENIUM_WEBDRIVER_PORT = '' + (config.containerPortMappings[config.sidecarImage][0]?.containerPort ?: '')
```

Comment:
Why do we need a `[0]` here? Isn't the structure `[imageName: [key: value, ..], ..]`?

Reply:
I don't think so, the API allows to specify multiple port mappings, so it's an array.

E.g. in the `config.yml`, I could define:

```yaml
steps:
  dockerExecute:
    containerPortMapping:
        - myImage:
            - '4444:4444'
            - '8080:8000'
```

The assumption here obviously is that there's only one port mapping defined for the sidecar.

Reply:
Ok, you're right.